File: 004/triangle.py
def check(a, b, c):
    if a + b > c or a + c > b or b + c > a:
        return True
    else:
        return False

def type(a, b, c):
    if a == b == c:
        return "Tam giac deu"
    elif a == b or a == c or b ==c:
        return "Tam giac can"
    elif a**2 + b**2 == c**2 or a**2 + c**2 == b**2 or b**2 + c**2 == a**2:
        return "Tam giac vuong"
    else:
        return "Tam giac thuong"

def Heron(a, b, c):
    p = (a + b + c)/2
    return (p*(p - a)*(p - b)*(p - c))**0.5

File: 004/triangle.java
public class triangle {
    public static boolean check(double a, double b, double c) {
        return (a + b > c) && (a + c > b) && (b + c > a);
    }

    public static String type(double a, double b, double c) {
        if (!check(a, b, c)) {
            return "Khong phai tam giac";
        }

        boolean vuong = Math.abs(Math.pow(a, 2) + Math.pow(b, 2) - Math.pow(c, 2)) < 1e-6 ||
                          Math.abs(Math.pow(b, 2) + Math.pow(c, 2) - Math.pow(a, 2)) < 1e-6 ||
                          Math.abs(Math.pow(a, 2) + Math.pow(c, 2) - Math.pow(b, 2)) < 1e-6;

        boolean deu = (a == b && b == c);
        boolean can = (a == b || a == c || b == c);

        if (deu) {
            return "Tam giac deu";
        }

        if (vuong && can) {
            return "Tam giac vuong can";
        }

        if (can) {
            return "Tam giac can";
        }

        if (vuong) {
            return "Tam giac vuong";
        }

        return "Tam giac thuong";
    }

    public static double Heron(double a, double b, double c) {
        double p = (a + b + c) / 2;
        return Math.sqrt(p * (p - a) * (p - b) * (p - c));
    }
}

